hedyorg/hedy
# 💻 show some roles of variables (constant, input constant, walker, container)

Adds roles in the variable list, works for four roles now, constant, input constant, container and walker. 

**How to test**

Follow these steps to verify this PR works as intended:

* change the feature flag in debugging.ts called showRoles to true
* check that the variable list now shows roles of variables, in addition to the variable names and their values. (when using the Hedy debugger)
* run this example program in level 2 and see if it says naam is a constant
```
naam is Hedy
leeftijd is 10
leeftijd is 15
print naam is leeftijd jaar oud
```
* see if in level 2 the variables in this program are input constants
```
naam is ask Wat is je naam?
print Hoi naam
leeftijd is ask Hoe oud ben je?
print naam is leeftijd jaar oud
```
* run this example program in level 10 and see if it says that dier is a walker
```
dieren = hond, kat, papegaai
voor dier in dieren
  print dier ' is een leuk dier!'
```
* go to level 3 and see if dieren is a container in this example code
```
dieren is hond, kat, papegaai
print dieren at random
```
* change the feature flag back to false
* check that the variable list now looks like it normally does, with only the variable names and their values.


**Checklist**
Done? Check if you have it all in place using this list: (mark with x if done)

- [x ] Contains one of the PR categories in the name
- [x ] Describes changes in the format above
- [x ] Has a "How to test" section

If you're unsure about any of these, don't hesitate to ask. We're here to help!
